src/app/clusters/operational-state-server/operational-state-server.h

```diff
@@ -0,0 +1,210 @@
+#include <app/clusters/operational-state-server/operational-state-delegate.h>
+#include <app/util/af.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+
+class Uncopyable
+{
+protected:
+    Uncopyable() {}
+    ~Uncopyable() {}
+
+private:
+    Uncopyable(const Uncopyable &);
+    Uncopyable & operator=(const Uncopyable &);
+};
+
+class OperationalStateServer : public CommandHandlerInterface, public AttributeAccessInterface, public Uncopyable
+{
+
+public:
+    template <typename T>
+    using DataModelListTemplate          = app::DataModel::List<T>;
+    using OperationalStateStructType     = app::Clusters::OperationalState::Structs::OperationalStateStruct::Type;
+    using OperationalStateStructTypeList = app::DataModel::List<OperationalStateStructType>;
+    using PhaseList                      = chip::app::DataModel::List<const chip::CharSpan>;
+    /**
+     * Init the operational state server.
+     * @param void
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    CHIP_ERROR Init();
+
+    // Inherited from CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+
+    /// IM-level implementation of read
+    ///
+    /// Returns appropriately mapped CHIP_ERROR if applicable (may return CHIP_IM_GLOBAL_STATUS errors)
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+    /**
+     * Set operational state list.
+     * @param operationalStateList The operational state list for which to save.
+     *  Template for save operational state alias cluster
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    template <typename T>
+    CHIP_ERROR SetOperationalStateList(const DataModelListTemplate<T> & operationalStateList);
+
+    /**
+     * Get operational state list.
+     * @param operationalStateList The pointer to save operational state list.
+     * @param size The number of operational state list's item.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    CHIP_ERROR GetOperationalStateList(OperationalStateStructDynamicList ** operationalStateList, size_t & size);
+
+    /**
+     * Rlease OperationalStateStructDynamicList
+     * @param operationalStateList The pointer for which to clear the OperationalStateStructDynamicList.
+     * @return void
+     */
+    void ReleaseOperationalStateList(OperationalStateStructDynamicList * operationalStateList);
+
+    /**
+     * Clear operational state list.
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    CHIP_ERROR ClearOperationalStateList();
+
+    /**
+     * Set phase list.
+     * @param phaseList The phase list for which to save.
```

Comment:
See above comments for state list; they all apply to phase list.

Reply:
Done.